### PR TITLE
Custom cdn path.

### PR DESCRIPTION
**Pull request by galachad**
Allow to set up custom CDN path for accessing the dependency resources.

### CODE REVIEW

File: R/semanticPage.R [hunk 1]
```diff
@@ -5,8 +5,14 @@ SUPPORTED_THEMES <- c("cerulean", "darkly", "paper", "simplex",
                       "readable",  "united", "journal", "solar",
                       "cyborg", "sandstone", "yeti", "lumen", "spacelab")
 
-#' Cloudfront path
-CDN_PATH <- "https://d335w9rbwpvuxm.cloudfront.net"
+#' Get CDN path semantic dependencies
+#'
+#' Internal function that return cdn path from options.
```
Comment (dokato) [inline R/semanticPage.R]: typo "returns"

Comment (dokato) [reply R/semanticPage.R]: ...and CDN in capitals

File: R/semanticPage.R [hunk 2]
```diff
@@ -5,8 +5,14 @@ SUPPORTED_THEMES <- c("cerulean", "darkly", "paper", "simplex",
...
+#'
+#' @return CDN path of semantic dependencies
```
[the remark this answers is not in the document]
Comment (galachad) [reply R/semanticPage.R]: Would you like to see this information in README?

[the remark this answers is not in the document]
Comment (dokato) [reply R/semanticPage.R]: Nope. Description of this function is enough.

File: tests/testthat/test_semanticPage.R [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+test_that("test get_cdn_path return default path", {
+  # When
+  options("shiny.custom.semantic.cdn" = NULL)
+  # Then
+  expect_equal(get_cdn_path(), "https://d335w9rbwpvuxm.cloudfront.net")
+})
+
+test_that("test get_cdn_path return path", {
+  # When
+  options("shiny.custom.semantic.cdn" = "shiny.semantic")
```
Comment (marekrogala) [inline tests/testthat/test_semanticPage.R]: nit: I think the value in the test is misleading, instead I suggest using some fake url for custom cdn path

**Comment by marekrogala** [comment]
Looks good to me

File: tests/testthat/test_semanticPage.R (hunk 3 above)
Comment (galachad) [reply tests/testthat/test_semanticPage.R]: `shiny.semantic` is valid URL when you would like to serve files locally from shiny. In shiny 1.3 the speed of hosting static files was improved dramatically, so we can even think about move semantic dependencies to be part of the code.

**Pull request merged**